ZEP/instructions/zep0000.md

```diff
@@ -0,0 +1,399 @@
+# Zarr Community Feedback Process - Zarr Enhancement Proposal (ZEP)
+
+--- 
+ 
+## ZEP 0 — Purpose and process
+
+Author: Sanket Verma <svsanketverma5@gmail.com>
+
+Status: Active
+
+Type: Process
+
+Created: 2022-14-03
+
+## What is ZEP?
+
+ZEP stands for Zarr Enhancement Proposal. A ZEP is a design document providing
+information to the Zarr community, describing a modification or enhancement of
+the Zarr storage specification, a new feature for its processes or environment.
+The ZEP should provide specific proposed changes to the Zarr specification and
+a narrative rationale for the specification changes.
+
+We intend ZEPs to be the primary mechanism for evolving the spec, collecting
+community input on major issues and documenting the design decision that has
+gone into Zarr. In addition, the ZEP author is responsible for building
+consensus within the community and documenting dissenting opinions.
+
+Because the ZEPs are maintained as text files in a versioned repository, their
+revision history is the historical record of the feature proposal.
+
+## ZEP audience
+
+The typical primary audience for ZEPs is the developers working on Zarr and its
+various implementations, the Zarr steering council as well as the Zarr
+community.
+
+The broader Zarr community may also choose to use the process to document
+expected feature additions and to manage complex design coordination problems
+that require collaboration across multiple projects.
+
+## ZEP types
+
+> Specification ZEP
+
+Specification ZEPs deal with the changes related to [Zarr Specifications].
+The SPEC ZEP (abbreviation of Specification ZEP) introduces changes in the core
+specification of Zarr. The changes related to core specification follow a
+strict and thorough review process and should be adopted by everyone in the
+Zarr community.
+
+- Core protocol ZEP
+
+Describes a ZEP which involves changes in the core specification of Zarr. Core
+protocol ZEPs (commonly known as Core ZEPs) are a part of SPEC ZEPs and apply to
+[Zarr Specifications] and its various implementations under the
+[zarr_implementations] GitHub repo. The core protocol ZEPs should be adopted by
+every implementation of Zarr and, in general, the overall Zarr community. Core
+ZEPs must go through a thorough review process, including involvement,
+discussion, and voting from the author of the proposed ZEP, Zarr Steering
+Council, author(s) of various Zarr implementations, and open-source
+projects/research groups using Zarr and the general Zarr community. The general
+advice is that everyone should be made aware of changes introduced by core
+ZEPs. Core protocol ZEPs require community consensus, and consumers are typically not
```

Comment:
Can we clarify what is meant by "consumers"? That word has not been part of our lexicon so far.

Reply:
Fair enough. Sorry it was a late change. It was previously "users" which I don't think fits. "Developers"?

Reply:
I think 'users' refers to a broader community where it involves contributors, maintainers and users of the project whereas Zarr developers would mean contributors and maintainers only.

How about using 'Zarr developers' and 'users' both?

Reply:
Ideally Zarr users don't really have to know much about the spec. Maybe we need the notion of "downstream users" or "dependents of" zarr? Or maybe we just stick with "developers".

Reply:
> Ideally Zarr users don't really have to know much about the spec.

I just wanted the ZEP process to be inclusive and welcoming for everyone and not leave out any part(s) of the Zarr community while designing/discussing a ZEP.

I've added this at the start:

> WHERE:
> 
> - Developers refer to contributors and maintainers of the project
> - User(s) refers to an individual or group of individuals or the broader
> community using the project in any way.

and used the term **developers** and **users** instead of *consumers*.

Let me know what you think of this. ^

Reply:
What about Zarr implementers?

Reply:
I think implementers come under the term 'developer'. Developer is more like an umbrella terminology here.